Libraries/src/Amazon.Lambda.Annotations.SourceGenerators/LambdaFunctionCodeGenerator.cs

```diff
@@ -0,0 +1,261 @@
+
+        public LambdaFunctionCodeGenerator(
+            MethodDeclarationSyntax lambdaFunctionMethodSyntax,
+            ClassDeclarationSyntax startupSyntax,
+            GeneratorExecutionContext context)
+        {
+            _lambdaFunctionMethodSyntax = lambdaFunctionMethodSyntax;
+            _startupSyntax = startupSyntax;
+            _context = context;
+        }
+
+        /// <summary>
+        /// Generates <see cref="SourceText"/> for the LambdaFunction including setting up the dependency injection.
+        /// </summary>
+        /// <returns>A <see cref="Tuple"/> containing source file name hint and source text.</returns>
+        public (string, SourceText) GenerateSource()
+        {
+            var apiGatewayResponseSymbol =
+                _context.Compilation.GetTypeByMetadataName("Amazon.Lambda.APIGatewayEvents.APIGatewayProxyResponse");
+
+            var returnTypeString = "APIGatewayProxyResponse";
+            if (LambdaFunctionMethodSymbol.IsAsync)
+            {
+                returnTypeString = $"System.Threading.Tasks.Task<{returnTypeString}>";
+            }
+
+            var source = new StringBuilder(
+$@"using System;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.Extensions.DependencyInjection;
+using System.Collections.Generic;
+
+namespace {LambdaFunctionMethodSymbol.ContainingNamespace}
+{{
+    public class {ClassName}
+    {{
+        private readonly ServiceProvider serviceProvider;
+
+        public {ClassName}()
+        {{
+            var services = new ServiceCollection();
+
+            // By default, Lambda function class is added to the service container using the scoped lifetime
+            // because web dependencies are normally scoped to the client request. To use a different lifetime,
+            // specify the lifetime in Startup.ConfigureServices(IServiceCollection) method.
+            services.AddScoped<{LambdaFunctionMethodSymbol.ContainingType}>();");
+
+            // Call Configure method if LambdaStartup is provided
+            if (ConfigureMethodSymbol != null)
+            {
+                source.Append($@"
+            var startup = new {ConfigureMethodSymbol.ContainingType}();
+            startup.{ConfigureMethodSymbol.Name}(services);");
+            }
+
+            source.Append($@"
+            serviceProvider = services.BuildServiceProvider();
+        }}
+
+        public{(LambdaFunctionMethodSymbol.IsAsync ? " async " : " ")}{returnTypeString} {LambdaFunctionMethodSymbol.Name}(APIGatewayProxyRequest request, ILambdaContext _context)
+        {{
+            // Create a scope for every request,
+            // this allows creating scoped dependencies without creating a scope manually.
+            using var scope = serviceProvider.CreateScope();
+
+            var {LambdaFunctionMethodSymbol.ContainingType.Name.ToCamelCase()} = scope.ServiceProvider.GetRequiredService<{LambdaFunctionMethodSymbol.ContainingType}>();");
```

Comment:
If I understand this would mean the constructor of the type containing the LambdaFunction would be called for every request. I think that would be a big break to the programming model for Lambda. I was thinking the scope would be for each invocation of the method. User's could use the `FromService` attribute on the method to get scoped services otherwise use the constructor for singleton services.

Reply:
That's right, customer code is like a ASP.NET Web API controller which is initialized for every request (it is a scoped dependency like the comment mention in L156).

The generated function follows the current Lambda function model, where constructor is called only once during the invocation and then stays in memory as long as Lambda wants it. We are getting best of both worlds here IMO Lambda + ASP.NET Core.

That said, if customer ever want to change the behavior of their Function (attributed with LambdaFunction), they can configure the lifetime of Function class in LambdaStartup.